### PR TITLE
OCM-2376 | ci : add script to run in prow job to validate commit msgs

### PR DESCRIPTION
**Pull request by ciaranRoche**
Adding a script and make target to be run as part of a new prow job to validate commit message hygiene

### CODE REVIEW

File: hack/commit-msg-verify.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+for sha in $commits; do
+  message=$(git log --format=%B -n 1 $sha)
+  if ! [[ $message =~ $pattern ]]; then
+    echo "Invalid commit message: $message"
+    echo "Expected format: JIRA_TICKET | TYPE : MESSAGE"
+    exit 1
+  fi
+done
+
+echo "All commit message are valid"
```
Comment (hunterkepley) [inline hack/commit-msg-verify.sh:20]: NP: newline at end of file

File: hack/commit-msg-verify.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+# PULL_BASE_SHA and PULL_PULL_SHA are set in prow job
+PULL_BASE_SHA="${PULL_BASE_SHA:-$(git merge-base master HEAD)}"
+PULL_PULL_SHA="${PULL_PULL_SHA:-HEAD}"
+
+pattern="^[A-Z]+-[0-9]+ \| (feat|fix|docs|style|refactor|test|chore|build|ci|perf) : .*$"
```
Comment (hunterkepley) [inline hack/commit-msg-verify.sh:7]: Will we have a document/announcement about this format?

**Review by hunterkepley — COMMENTED** [review]
overall LGTM besides the newline. Will approve once that is added

File: hack/commit-msg-verify.sh (hunk 2 above)
Comment (ciaranRoche) [reply hack/commit-msg-verify.sh:7]: we already do, are you telling me do didnt read our Contribution guide? TUT TUT TUT

https://github.com/openshift/rosa/blob/master/CONTRIBUTING.md

Also this is why we need this job, no one reads the documentation :rofl:

Comment (hunterkepley) [reply hack/commit-msg-verify.sh:7]: :laughing:  i am ashamed

**Pull request merged**